### PR TITLE
Add loggers, defensive checks and  tests to `Storage`

### LINKED ISSUE #149: [Task] Add loggers, more defensive checks and tests to `Storage`
## Describe the Task
Storage works as intended, but more safety checks and descriptive logs should be added.
This helps aid in integration with tokenizers down the road and could pre-emptively catch any developer errors.

## Expected behaviour
- More testing and defensive checks for edgecases (such as `null` inputs)
- Loggers for thrown exceptions

### PR DESCRIPTION
**Pull request by nichyjt**
This closes #149.

## Changelog
- Added assertions in most methods in `Storage` to make execution more watertight
- Added tests to catch developer abuse of `save` and `load` functionality
- Added logic guards to ensure that only allowed fileNames can have access to IO functions
- Added tests to validate logic guards work

### CODE REVIEW

File: src/main/java/wellnus/storage/Storage.java [hunk 1]
```diff
@@ -87,6 +122,12 @@ private void verifyDataDirectory() throws StorageException {
      * @param fileName data file to retrieve
      */
     protected File getFile(String fileName) throws StorageException {
+        assert fileName != null : ASSERT_FILENAME_NOT_NULL;
+        assert fileName.length() > 0 : ASSERT_FILENAME_NOT_EMPTY;
```
Comment (wenxin-c) [inline src/main/java/wellnus/storage/Storage.java:126]: Maybe can change ```0``` to a constant?

File: src/main/java/wellnus/storage/Storage.java [hunk 2]
```diff
@@ -228,6 +284,13 @@ private String loadDataFromDisk(File file) throws StorageException {
      */
     //@@author nichyjt
     public void saveData(ArrayList<String> tokenizedManager, String fileName) throws StorageException {
+        assert fileName != null : ASSERT_FILENAME_NOT_NULL;
+        assert fileName.length() > 0 : ASSERT_FILENAME_NOT_EMPTY;
```
Comment (wenxin-c) [inline src/main/java/wellnus/storage/Storage.java:288]: Same here ```0``` can be changed to a constant, and line 314, 333

File: src/test/java/wellnus/storage/StorageTest.java [hunk 3]
```diff
@@ -130,4 +132,83 @@ public void saveAndLoadData_test() {
...
+        try {
+            storage.loadData(Storage.FILE_DEBUG);
+        } catch (StorageException exception) {
+            fail("loadData failed when loading file that does not exist despite safety checks");
+        }
+        // Cleanup the debug file that was created as part of safety measures
+        // deleteFile must work as the above tests on deleteFile have passed
+        try {
+            storage.deleteFile(Storage.FILE_DEBUG);
+        } catch (StorageException e) {
```
Comment (wenxin-c) [inline src/test/java/wellnus/storage/StorageTest.java:168]: try not to use ```e``` as exception name

**Review by wenxin-c — COMMENTED** [review]
Overall is good, just a little bit magic literal suggestions. After fixing LGTM!!

**Review by wenxin-c — APPROVED** [review]
LGTM!!
